cmd/result-aggregator/htpasswd.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+)
+
+// lookup passwords in a htpasswd file
+// The entries must have been created with -s for SHA encryption
+
+type HtpasswdFile struct {
+	Users map[string]string
+}
+
+func NewHtpasswdFromFile(path string) (*HtpasswdFile, error) {
+	r, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Close()
+	return NewHtpasswd(r)
+}
+
+func NewHtpasswd(file io.Reader) (*HtpasswdFile, error) {
+	csvReader := csv.NewReader(file)
+	csvReader.Comma = ':'
+	csvReader.Comment = '#'
+	csvReader.TrimLeadingSpace = true
+
+	records, err := csvReader.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	h := &HtpasswdFile{Users: make(map[string]string)}
+
+	for _, record := range records {
+		if len(record) < 2 {
+			return nil, fmt.Errorf("abnormal content of Htpasswd File")
+		}
+		h.Users[record[0]] = record[1]
+	}
+	return h, nil
+}
+
+func (h *HtpasswdFile) Validate(user string, password string) bool {
+	realPassword, exists := h.Users[user]
+	if !exists {
+		return false
+	}
+	return realPassword == password
```

Comment:
The password in a htpasswd file is hashed, you can not compare it like this, you need to do something like `bcrypt.CompareHashAndPassword` after figuring out the hash format in the htpasswd file, not sure how to do that